apps/api/src/app/content-templates/usecases/compile-template/email/compile-email-template.usecase.ts

```diff
@@ -0,0 +1,152 @@
+import { IEmailBlock, OrganizationRepository, OrganizationEntity } from '@novu/dal';
+import { CompileTemplate } from '../compile-template.usecase';
+import { CompileTemplateCommand } from '../compile-template.command';
+import { ApiException } from '../../../../shared/exceptions/api.exception';
+import * as fs from 'fs';
+import { CompileEmailTemplateCommand } from './compile-email-template.command';
+import { GetLayoutCommand, GetLayoutUseCase } from '../../../../layouts/use-cases';
+import { LayoutId } from '@novu/shared';
+
+const helperContent = fs.readFileSync(
+  `${__dirname}${
+    !process.env.E2E_RUNNER ? '/src/app/content-templates/usecases/compile-template' : '/../../compile-template'
+  }/templates/basic.handlebars`,
+  'utf8'
+);
+
+const defaultLayout = fs.readFileSync(
+  `${__dirname}${
+    !process.env.E2E_RUNNER ? '/src/app/content-templates/usecases/compile-template' : '/../../compile-template'
+  }/templates/layout.handlebars`,
+  'utf8'
+);
+
+@Injectable()
+export class CompileEmailTemplateUsecase {
+  constructor(
+    private compileTemplate: CompileTemplate,
+    private organizationRepository: OrganizationRepository,
+    private getLayoutUsecase: GetLayoutUseCase
+  ) {}
+
+  public async execute(command: CompileEmailTemplateCommand) {
+    const organization: OrganizationEntity | null = await this.organizationRepository.findById(command.organizationId);
+    if (!organization) throw new NotFoundException(`Organization ${command.organizationId} not found`);
+    let useNovuDefault = !command.layoutId;
+    let layout;
+
+    if (!useNovuDefault) {
+      try {
+        layout = await this.getLayoutUsecase.execute(
+          GetLayoutCommand.create({
+            layoutId: command.layoutId as LayoutId,
+            environmentId: command.environmentId,
+            organizationId: command.organizationId,
+          })
+        );
+      } catch (e) {
+        useNovuDefault = true;
+      }
+    }
+
+    const isEditorMode = command.contentType === 'editor';
+    let subject = '';
+    let content: string | IEmailBlock[] = command.content;
+    let preheader = command.preheader;
+    const layoutContent = useNovuDefault ? defaultLayout : layout.content;
+
+    const payload = {
+      ...command.payload,
+      subject: command.subject,
+      preheader,
+      blocks: [],
+      branding: {
+        logo: organization.branding?.logo,
+        color: organization.branding?.color || '#f47373',
+      },
+    };
+
+    try {
+      subject = await this.renderContent(command.subject, payload);
+      if (isEditorMode) {
+        content = await this.getContent(content, payload);
+      }
+      if (preheader) {
+        preheader = await this.renderContent(preheader, payload);
+      }
+    } catch (e) {
+      throw new ApiException(e?.message || `Message content could not be generated`);
+    }
+
+    const customTemplate = CompileEmailTemplateUsecase.addPreheader(layoutContent as string);
```

Comment:
won't this preheader be included twice for the default layout?

Reply:
Yes. But I am now changing the logic of where the default layout will be. So it will not be relevant anymore :) The novu default would not get here